src/plugins/intel_cpu/src/utils/shape_inference/shape_inference.cpp

```diff
@@ -357,6 +362,45 @@ class ShapeInferTA : public ShapeInferBase {
     }
 };
 
+template <class TOp, uint32_t MASK>
+class ShapeInferenceTA : public ShapeInferBase {
+public:
+    using ShapeInferBase::ShapeInferBase;
+
+    IShapeInferCommon::Result infer(const std::vector<StaticShape>& input_shapes,
+                                    const ov::ITensorAccessor& tensor_accessor) override {
+        // Temporary support of StaticShape.
+        auto in_shapes = std::vector<StaticShapeRef>(input_shapes.size());
+        std::transform(input_shapes.begin(), input_shapes.end(), in_shapes.begin(), [](const StaticShape& s) {
+            return StaticShapeRef(reinterpret_cast<const VectorDims&>(s));
+        });
+
+        auto out_shapes = infer(in_shapes, tensor_accessor);
+        Result result{{}, out_shapes ? ShapeInferStatus::success : ShapeInferStatus::skip};
+
+        if (out_shapes) {
+            result.shapes.reserve(out_shapes->size());
+            std::transform(out_shapes->begin(),
+                           out_shapes->end(),
+                           std::back_inserter(result.shapes),
+                           [](StaticShapeCon& s) {
+                               return std::move(reinterpret_cast<StaticShape&&>(*s));
+                           });
+        }
+
+        return result;
+    }
+
+    ov::optional<std::vector<StaticShapeCon>> infer(const std::vector<StaticShapeRef>& input_shapes,
```

Comment:
Could you elaborate more about advantages of `optional` usage here?
Is there a case when the returned output shape is not needed, like inputs validation only?

Reply:
As the `infer` interface function is modified (different input types and return type), I introduced to use optional to replace the `IShapeInferCommon::Result` which do similar thing but using different approach.

This result struct  returned from `infer` in most cases is {output_shapes, success}, the optional do same thing return valid data. But there are some cases like in fallback path or in CPU specific implementation when this status can be `skip` and initialize vector with invalid data (no data).

The benefit of this optional is that `output_shapes` there is dummy vector initialization in case of invalid data and can be done later if required. In case of using `IShapeInferCommon::Result` or std::pair<T, bool> this vector will be always initialized.